src/app/public/modules/back-to-top/back-to-top-adapter.service.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  ElementRef,
+  Injectable,
+  OnDestroy
+} from '@angular/core';
+
+import {
+  SkyAppConfig
+} from '@skyux/config';
+
+import {
+  SkyAppWindowRef
+} from '@skyux/core';
+
+import {
+  Observable
+} from 'rxjs/Observable';
+
+import {
+  Subject
+} from 'rxjs/Subject';
+
+import 'rxjs/add/observable/fromEvent';
+
+import 'rxjs/add/operator/filter';
+
+@Injectable()
+export class SkyBackToTopDomAdapterService implements OnDestroy {
+
+  private ngUnsubscribe = new Subject<void>();
+
+  constructor(
+    private skyAppConfig: SkyAppConfig,
+    private windowRef: SkyAppWindowRef
+  ) { }
+
+  public ngOnDestroy(): void {
+    this.ngUnsubscribe.next();
+    this.ngUnsubscribe.complete();
+  }
+
+  /**
+   * This event returns a boolean on scroll indicating whether the provided element is in view.
+   * @param elementRef The target element reference.
+   */
+  public elementInViewOnScroll(elementRef: ElementRef): Observable<boolean> {
+    const parent = this.findScrollableParent(elementRef.nativeElement);
+
+    return Observable
+      .fromEvent(parent, 'scroll')
+      .takeUntil(this.ngUnsubscribe)
+      .map(() => {
+        const isInView = this.isElementScrolledInView(
+          elementRef.nativeElement,
+          parent
+        );
+        return isInView;
+    });
+  }
+
+  /**
+   * Scrolls the window or scrollable parent to the provided element.
+   * @param elementRef The target element reference.
+   */
+  public scrollToElement(elementRef: ElementRef): void {
+    /* sanity check */
+    /* istanbul ignore if */
+    if (!elementRef || !elementRef.nativeElement) {
+      return;
+    }
+
+    const windowObj = this.windowRef.nativeWindow;
+    const parent = this.findScrollableParent(elementRef.nativeElement);
+
+    if (parent === windowObj) {
+      // Scroll to top, but account for omnibar if it exists.
+      const omnibar = this.skyAppConfig.skyux.omnibar;
```

Comment:
Pulling `SkyAppConfig` into the DOM adapter doesn't feel right (business logic versus UI). Since we're in a "DOM" adapter, could you simply query to see if `.sky-omnibar-iframe` exists?